web/modules/custom/dpl_opening_hours/dpl_opening_hours.module

```diff
@@ -47,6 +52,79 @@ void {
   }
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter() for the taxonomy term deletion form related to opening hours.
+ *
+ * This hook modifies the deletion confirmation form for taxonomy terms within
+ * the "opening_hours_categories" vocabulary, adding custom messages.
+ *
+ * @param array $form
+ *   The form structure array.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ * @param string $form_id
+ *   The ID of the form being altered.
+ */
+function dpl_opening_hours_form_taxonomy_term_opening_hours_categories_delete_form_alter(array &$form, FormStateInterface $form_state, string $form_id): void {
+  $form_object = $form_state->getFormObject();
+  if (!$form_object instanceof EntityFormInterface) {
+    return;
+  }
+
+  $term = $form_object->getEntity();
+  if (!$term instanceof Term) {
+    return;
+  }
+
+  $term_id = $term->id();
+  if (!is_int($term_id) && !is_string($term_id)) {
+    return;
+  }
+
+  $term_label = $term->label();
+
+  $form['#title'] = t('Are you sure you want to delete all "@term_label" opening hours?', [
+    '@term_label' => $term_label,
+  ]);
+
+  $opening_hours_repository = DrupalTyped::service(OpeningHoursRepository::class, 'dpl_opening_hours.repository');
+  $opening_hours_in_category = $opening_hours_repository->loadMultiple(categoryId: (int) $term_id);
+  $opening_hours_count = count($opening_hours_in_category);
+
+  $form['description']['#markup'] = t('@term_label is associated with @count opening hours. Deleting @term_label will also delete these opening hours. Are you sure you want to continue?', [
+    '@term_label' => $term_label,
+    '@count' => $opening_hours_count,
+  ]);
+}
+
+/**
+ * Implements hook_taxonomy_term_predelete().
+ *
+ * Deletes associated opening hours when a category term is deleted.
+ *
+ * @param \Drupal\taxonomy\TermInterface $term
+ *   The term being deleted.
+ */
+function dpl_opening_hours_taxonomy_term_predelete(TermInterface $term): void {
+  if ($term->bundle() !== 'opening_hours_categories') {
+    return;
+  }
+
+  $term_id = $term->id();
+  if (!is_int($term_id) && !is_string($term_id)) {
+    return;
+  }
+
+  $opening_hours_repository = DrupalTyped::service(OpeningHoursRepository::class, 'dpl_opening_hours.repository');
+  $opening_hours_in_category = $opening_hours_repository->loadMultiple(categoryId: (int) $term_id);
+
+  foreach ($opening_hours_in_category as $instance) {
+    if ($instance->id && $instance->repetition->id) {
```

Comment:
```suggestion
    if ($instance->id) {
```

I do not think you need the check for the repetition id.

The `delete()` method for the repository allows the repetition id to be `null` so we should be able to pass this value as well if the repetition of the instance does not have an id.